### PR TITLE
[Relay] Add LowerTEPass into vm lowering proecss

### PR DESCRIPTION
**Pull request by mikepapadim**
This PR adds the `LowerTEPass`  in the VM compiler.

@mbs-octoml  @jroesch @electriclilies

### CODE REVIEW

**Comment by Mousius** [comment]
Hi @mikepapadim, my apologies but I don't quite understand what the impact of this change is as it appears to have no effect on any of the test cases - could you add some information to motivate the change?

**Comment by electriclilies** [comment]
Just catching up after being out for a week -- is "vm_mod" used for anything other than name mangling? 
Also is LowerTEPass supposed replacing any existing lowering logic in the VM? (or further downstream?)

File: src/relay/backend/vm/compiler.cc [hunk 1]
```diff
@@ -1083,13 +1082,21 @@ IRModule VMCompiler::OptimizeModule(IRModule mod, const TargetsMap& targets_arg,
   pass_seqs.push_back(transform::InferType());
   pass_seqs.push_back(transform::LabelOps());
 
+  tec::TargetMap targets_;
+  tec::DeviceMap device_map;
+
+  targets_ = GetTecTargetMapFromTargetMap(targets);
+
+  pass_seqs.push_back(tec::LowerTEPass(targets_, device_map, "vm_mod", [this](Function func) {}));
```
Comment (mbs-octoml) [inline src/relay/backend/vm/compiler.cc:1090]: For others: This PR also needs to remove the direct calls to the TECompiler Lower and LowerShapeFunc which are a bit fiddly to change over. I believe Michalis is working on that.

**Comment by masahi** [comment]
What's the status of this PR?

**Comment by mbs-octoml** [comment]
@masahi this ended up being subsumed by #9613, I think it can be closed.

**Pull request closed without merging**